plugin.js

```diff
@@ -281,24 +286,64 @@ function processWorklets(t, path, processor) {
   }
 }
 
+const PLUGIN_BLACKLIST_NAMES = [
+  // '@babel/plugin-transform-destructuring', do not blacklist
+  'metro-react-native-babel-preset/node_modules/@babel/plugin-transform-object-assign',
+];
+
+const PLUGIN_BLACKLIST = PLUGIN_BLACKLIST_NAMES.map((pluginName) => {
+  try {
+    const blacklistedPluginObject = require(pluginName);
+    const blacklistedPlugin = blacklistedPluginObject.default({
```

Comment:
this seem to be very obscure and rely on undocumented babel features. Can we add some comments on what it does as when the babel api changes we won't be able to track down what this piece of code should do